src/fsharp/TypeChecker.fs

```diff
@@ -732,30 +732,41 @@ let UnifyUnitType cenv denv m ty exprOpt =
             if not (typeEquiv cenv.g cenv.g.bool_ty ty) then 
                 warning (UnitTypeExpected (denv,ty,m)) 
             else
-                match exprOpt with 
-                | Some(Expr.App(Expr.Val(vf,_,_),_,_,exprs,_)) when vf.LogicalName = opNameEquals ->
-                    match exprs with 
-                    | Expr.App(Expr.Val(propRef,_,_),_,_,Expr.Val(vf,_,_) :: _,_) :: _ ->
-                        if propRef.IsPropertyGetterMethod then
-                            let propertyName = propRef.PropertyName
-                            let hasCorrespondingSetter =
-                                match propRef.ActualParent with
-                                | Parent entityRef ->
-                                    entityRef.MembersOfFSharpTyconSorted
-                                    |> List.exists (fun valRef -> valRef.IsPropertySetterMethod && valRef.PropertyName = propertyName)
-                                | _ -> false
-
-                            if hasCorrespondingSetter then
-                                warning (UnitTypeExpectedWithPossiblePropertySetter (denv,ty,vf.DisplayName,propertyName,m))
+                let checkExpr exprOpt =
```

Comment:
Could you factor this out into a separate function that makes it obvious it is about error reporting please?  Thanks